contrail-kubernetes-master/hooks/contrail_kubernetes_master_hooks.py

```diff
@@ -39,10 +39,16 @@ def install():
     status_set("blocked", "Missing relation to contrail-controller")
 
 
+@hooks.hook("leader-elected")
+def leader_elected():
+    _notify_controller()
+
+
 @hooks.hook("config-changed")
 def config_changed():
     update_nrpe_config()
     _notify_contrail_kubernetes_node()
+    _notify_controller()
```

Comment:
condition can be here:

if config.changed("kubernetes_api_hostname") or config.changed("kubernetes_api_secure_port")